### PR TITLE
Service provider

### PR DESCRIPTION
**Pull request by deric**
Added support for changing service provider.

### CODE REVIEW

File: spec/classes/aerospike_spec.rb [hunk 1]
```diff
@@ -437,7 +453,10 @@
 				# Tests related to the aerospike::config class
 
 				# Tests related to the aerospike::service class
-				it { is_expected.to_not contain_service('amc') }
+				it { is_expected.to contain_service('aerospike').with_hasrestart(true)\
```
Comment (aerostitch) [inline spec/classes/aerospike_spec.rb:461]: Why dropping `it { is_expected.to_not contain_service('amc') }` ? It is important to keep tests that ensure that some services are not present in addition to test that some are present.

File: Gemfile [hunk 2]
```diff
@@ -4,12 +4,16 @@ group :test do
...
   gem "puppet", ENV['PUPPET_VERSION'] || '~> 3.8.4'
+  gem 'hiera'
   gem "rspec-core", '< 3.2' if RUBY_VERSION < '1.9'
   gem "puppet-lint"
   gem "rspec-puppet", :git => 'https://github.com/rodjek/rspec-puppet.git'
   gem "puppet-syntax"
   gem "puppetlabs_spec_helper"
   gem "metadata-json-lint"
+  if RUBY_VERSION < '2.2.5'
+    gem 'ruby_dep', '< 1.4'
```
Comment (aerostitch) [inline Gemfile]: This makes the tests fail on ruby 1.9.3.

File: Gemfile [hunk 3]
```diff
@@ -4,12 +4,16 @@ group :test do
   gem "json_pure", '2.0.1' if RUBY_VERSION < '2.0'
   gem "rake"
   gem "puppet", ENV['PUPPET_VERSION'] || '~> 3.8.4'
+  gem 'hiera'
```
Comment (aerostitch) [inline Gemfile]: Why do you add hiera? It's already a dependency of the puppet gem. It seems pointless to me.

File: Gemfile [hunk 4]
```diff
@@ -21,6 +25,8 @@ group :development do
 end
 
 group :development, :system_tests do
-  gem "beaker"
+  if RUBY_VERSION < '2.2.5'
+    gem 'beaker', '< 3'
```
Comment (aerostitch) [inline Gemfile]: can you use the oneliner notation instead. More compact and still really readable.

File: spec/classes/aerospike_spec.rb [hunk 5]
```diff
@@ -413,6 +413,22 @@
...
+        let(:facts) {{
+          :osfamily => osfamily,
+        }}
+
+        it { is_expected.to compile.with_all_deps }
+        it { is_expected.to contain_class('aerospike::install').that_comes_before('Class[aerospike::config]') }
+        it { is_expected.to contain_class('aerospike::config').that_comes_before('Class[aerospike::service]') }
+        it { is_expected.to contain_class('aerospike::service') }
+
+        it { is_expected.to contain_service('aerospike') }
```
Comment (aerostitch) [inline spec/classes/aerospike_spec.rb]: I don't see any test on the provider itself here. Maybe it misses a `.with_provider('systemd')` ?

**Review by aerostitch — CHANGES_REQUESTED** [review]

**Comment by aerostitch** [comment]
Hi @deric,

Thanks for your PR.
I added a few comments to address before merging.
Can you have a look please?

Thanks!
Joseph

**Comment by deric** [comment]
@aerostitch Sure, hopefully I've fixed all the comments. I've squashed commits into one, to remove the mess I've created. Sorry, I was doing this in a hurry.

**Review by aerostitch — APPROVED** [review]

**Comment by aerostitch** [comment]
Thanks a lot!

**Comment by aerostitch** [comment]
Damn I forgot to ask you to update the doc (README.md). Can you do another PR with the doc update (just add the parameter with the default value and a short description) if you have some time please?
If not, let me know I'll take care of it.
Thanks

**Comment by deric** [comment]
Yeah, no problem. There will be more PRs coming :)

**Comment by aerostitch** [comment]
perfect, thanks! :)

**Pull request merged**